connectors/sources/azure_blob_storage.py

```diff
@@ -37,17 +36,17 @@ class AzureBlobStorageDataSource(BaseDataSource):
     name = "Azure Blob Storage"
     service_type = "azure_blob_storage"
 
-    def __init__(self, configuration):
+    def __init__(self, configuration, pipeline):
         """Set up the connection to the azure base client
 
         Args:
             configuration (DataSourceConfiguration): Object of DataSourceConfiguration class.
         """
         super().__init__(configuration=configuration)
         self.connection_string = None
-        self.enable_content_extraction = self.configuration["enable_content_extraction"]
         self.retry_count = self.configuration["retry_count"]
         self.concurrent_downloads = self.configuration["concurrent_downloads"]
+        self.content_extraction_enabled = self.pipeline["extract_binary_content"]
```

Comment:
@artem-shelkovnikov using this draft to test ideas; is it okay to pass the pipeline over like this so it can access `extract_binary_content` or is there a better way to approach this?